src/java.base/share/classes/java/nio/channels/spi/AsynchronousChannelProvider.java

```diff
@@ -99,15 +92,7 @@ private static AsynchronousChannelProvider loadProviderAsService() {
                                    ClassLoader.getSystemClassLoader());
             Iterator<AsynchronousChannelProvider> i = sl.iterator();
             for (;;) {
-                try {
-                    return (i.hasNext()) ? i.next() : null;
-                } catch (ServiceConfigurationError sce) {
-                    if (sce.getCause() instanceof SecurityException) {
-                        // Ignore the security exception, try the next provider
-                        continue;
-                    }
-                    throw sce;
-                }
+                return (i.hasNext()) ? i.next() : null;
```

Comment:
The loop was only required to filter SecurityException so you change this to `return sl.findFirst().orElse(null);`